core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -35,7 +39,11 @@ export class ResultPanelComponent {
 
   private static readonly PRETTY_JSON_TEXT_LIMIT: number = 50000;
   private static readonly TABLE_COLUMN_TEXT_LIMIT: number = 1000;
-
+  public result :ReadonlyArray<ResultObject> | undefined;
+  public resultMap: Map<string, Result> = new Map<string, Result>();
+  public selectedOperatorID: string = "";
+  public chartType: string | undefined;
+  public showTable: boolean = true;
```

Comment:
try to reduce the number of class variables in a component. A general rule of thumb is that it keeps the minimal amount of data and states - it only keeps data it needs to display, and states that it need to control the display. For `showTable`, I think the HTML can check if `chartType` is undefined or not to decide showing chart vs table